lib/ansible/plugins/connection/ssh.py

```diff
@@ -44,6 +48,119 @@
 SSHPASS_AVAILABLE = None
 
 
+class AnsibleSshConnectionFailure(AnsibleConnectionFailure):
+    ''' the ssh connection plugin had a fatal error '''
+    pass
+
+
+# Could track retry attempts here. A little odd.
+class AnsibleAttemptedSshConnectionFailure(AnsibleConnectionFailure):
+    ''' the ssh connection plugin had a fatal error but may attempt to retry '''
+    pass
+
+class AnsibleSshInitialWriteConnectionFailure(AnsibleSshConnectionFailure):
+    ''' the ssh connection plugin had a fatal error during the initial data write '''
+    pass
+
+
+class AnsibleSshCommandErrorConnectionFailure(AnsibleSshConnectionFailure):
```

Comment:
these seem more valid, i would just drop the Ssh part and make them specific 'connection failure causes' but not per connection plugin